test/integration/gardenlet/managedseed/managedseed_test.go

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedseed_test
+
+import (
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/apis/seedmanagement/encoding"
+	seedmanagementv1alpha1 "github.com/gardener/gardener/pkg/apis/seedmanagement/v1alpha1"
+	gardenletconfigv1alpha1 "github.com/gardener/gardener/pkg/gardenlet/apis/config/v1alpha1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var _ = Describe("ManagedSeed controller test", func() {
+	var (
+		shoot             *gardencorev1beta1.Shoot
+		managedSeed       *seedmanagementv1alpha1.ManagedSeed
+		kubeconfigSecret  *corev1.Secret
+		testSecretBinding *gardencorev1beta1.SecretBinding
+		testSecret        *corev1.Secret
+
+		reconcileShoot = func() {
+			By("Patch the Shoot as Reconciled")
+			patch := client.MergeFrom(shoot.DeepCopy())
+			shoot.Status.ObservedGeneration = shoot.Generation
+			shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+				State: gardencorev1beta1.LastOperationStateSucceeded,
+			}
+			ExpectWithOffset(1, testClient.Status().Patch(ctx, shoot, patch)).To(Succeed())
+		}
+
+		deployGardenlet = func() {
```

Comment:
This function does not really deploy the gardenlet, or am I missing something? Shouldn't it rather be called `checkIfGardenletWasDeployed`?